common/djangoapps/student/helpers.py

```diff
@@ -368,6 +368,24 @@ def get_redirect_to(request):
     return redirect_to
 
 
+def _is_safe_redirect(request, redirect_to):
+    """
+    Determine if the given redirect URL/path is safe for redirection.
+
+    The redirect_to URL must be a safe URL. The redirect URL must also
+    be either the same domain or a subdomain of the request host domain.
+    """
+    request_host = request.get_host()
+    request_host_domain = '.'.join(request_host.split('.')[-2:])
+    redirect_to_host = urlparse.urlsplit(redirect_to).netloc
+    return (
+        # Use wildcard check to ensure redirect to the same domain or a subdomain
+        http.is_same_domain(urlparse.urlsplit(redirect_to).netloc, '.' + request_host_domain) and
```

Comment:
@douglashall: FYI: You had this as an "or" in the original POC, and I changed this to an "and".  I also changed the call to `is_safe_url` below to be a no-op when it came to testing allowed_hosts, but passing in the redirect_to_host.

Let me know if this makes sense to you.

Reply:
Sounds good if the end result is still the same.